src/txn/txn.c

```diff
@@ -2809,6 +2811,19 @@ __wt_verbose_dump_txn_one(
         break;
     }
 
+    memset(snapshot_buf, 0, sizeof(snapshot_buf));
+    strcat(snapshot_buf, "[");
+    for (i = 0; i < txn->snapshot_data.snapshot_count; i++) {
+        if (i == 0)
+            WT_RET(__wt_snprintf(
+              snapshot_buf_tmp, sizeof(snapshot_buf_tmp), "%lu", txn->snapshot_data.snapshot[i]));
+        else
+            WT_RET(__wt_snprintf(
+              snapshot_buf_tmp, sizeof(snapshot_buf_tmp), ", %lu", txn->snapshot_data.snapshot[i]));
+        strcat(snapshot_buf, snapshot_buf_tmp);
+    }
+    strcat(snapshot_buf, "]");
```

Comment:
Per the above comment users can modify `session_max` and there's no upper bound on the size of the `snapshot_buf` string. WiredTiger has a safe version of strcat that performs size checks on the destination buffer (`__wt_strcat`). Could you please change the code to use `__wt_strcat` and handle the error case when the buffer is too small?